src/chainhook/server.ts

```diff
@@ -0,0 +1,108 @@
+import * as fs from 'fs';
+import {
+  ChainhookEventObserver,
+  ChainhookNodeOptions,
+  Payload,
+  ServerOptions,
+  ServerPredicate,
+  StacksPayload,
+} from '@hirosystems/chainhook-client';
+import { PgStore } from '../pg/pg-store';
+import { ENV } from '../env';
+import { logger } from '@hirosystems/api-toolkit';
+import { randomUUID } from 'node:crypto';
+
+export function getPersistedPredicateFromDisk(): ServerPredicate | undefined {
+  const predicatePath = `${ENV.CHAINHOOK_PREDICATE_PATH}/predicate.json`;
+  try {
+    if (!fs.existsSync(predicatePath)) {
+      return;
+    }
+    const fileData = fs.readFileSync(predicatePath, 'utf-8');
+    return JSON.parse(fileData) as ServerPredicate;
+  } catch (error) {
+    logger.error(error, `ChainhookServer unable to get persisted predicate`);
+  }
+}
+
+export function persistPredicateToDisk(predicate: ServerPredicate) {
+  const predicatePath = `${ENV.CHAINHOOK_PREDICATE_PATH}/predicate.json`;
+  try {
+    try {
+      fs.mkdirSync(ENV.CHAINHOOK_PREDICATE_PATH);
+    } catch (error: any) {
+      if (error.code != 'EEXIST') throw error;
```

Comment:
I think this can be replaced with just `fs.mkdirSync(ENV.CHAINHOOK_PREDICATE_PATH, { recursive: true });`